test/worker_test.rb

```diff
@@ -744,6 +744,24 @@ def self.perform
     end
   end
 
+  it "reports errors correctly when an error occurs while pruning workers" do
```

Comment:
maybe the name of the test could be more explicit, like "correctly reports errors which don't have a backtrace"?